File: python/905.Sort-Array-By-Parity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        even = 0
        odd = len(nums) - 1

        while even < odd:
            if nums[even] % 2 != 0:
                nums[even], nums[odd] = nums[odd], nums[even]
                odd -= 1
            else:
                even += 1
        
        return nums

File: javascript/905.Sort-Array-By-Parity.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParity = function (nums) {
  even = nums.filter((n) => n % 2 === 0);
  odd = nums.filter((n) => n % 2 !== 0);

  return [...even, ...odd];
};
